### PR TITLE
705 raspberry port

### PR DESCRIPTION
**Pull request by vloginova**
Start porting Embox on Raspberry PI. These changes include essential hardware specific modules:
 - realization of the diag interface
 - interrupt controller
 - driver for the System Timer.

### CODE REVIEW

File: src/drivers/serial/raspi_uart.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#define UART0_FR     UART0_OFFSET + 0x18 /* Flag register */
+#define UART0_IBRD   UART0_OFFSET + 0x24 /* Integer Baud rate divisor */
+#define UART0_FBRD   UART0_OFFSET + 0x28 /* Fractional Baud rate divisor */
+#define UART0_LCRH   UART0_OFFSET + 0x2C /* Line Control register */
+#define UART0_CR     UART0_OFFSET + 0x30 /* Control register */
+#define UART0_IMSC   UART0_OFFSET + 0x38 /* Interrupt Mask Set Clear Register */
+#define UART0_ICR    UART0_OFFSET + 0x44 /* Interrupt Clear Register */
+
+
+static void delay(int32_t count) {
```
Comment (anton-bondarev) [inline src/drivers/serial/raspi_uart.c]: Assembler for delay routine in not very good idea. Can it be changed to more suit version?

File: src/drivers/clock/raspi_systick.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+#include <hal/clock.h>
+#include <hal/reg.h>
+#include <hal/system.h>
+#include <kernel/irq.h>
+#include <kernel/time/clock_source.h>
+
+#include <embox/unit.h>
+
+#define BCM2835_SYSTEM_TIMER_BASE 0x20003000
```
Comment (anton-bondarev) [inline src/drivers/clock/raspi_systick.c]: May be extract it as module option?

File: src/arch/arm/raspi/cpu_reset.S [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @file
+ * @brief
+ *
+ * @author  Vita Loginova
+ * @date    09.07.2015
+ */
+
+.global hardware_init_hook
+hardware_init_hook:
```
Comment (anton-bondarev) [inline src/arch/arm/raspi/cpu_reset.S]: It's a boot code?

**Comment by anton-bondarev** [comment]
LGTM

File: src/drivers/clock/raspi_systick.c (hunk 2 above)
Comment (vloginova) [inline src/drivers/clock/raspi_systick.c]: Why? It is a constant value for this chip. I don't think that there is a point to do that. At least for now.

File: src/arch/arm/raspi/cpu_reset.S (hunk 3 above)
Comment (vloginova) [inline src/arch/arm/raspi/cpu_reset.S]: Yes, see https://github.com/embox/embox/blob/master/src/arch/arm/armlib/head.S

File: src/drivers/serial/raspi_uart.c (hunk 1 above)
Comment (vloginova) [inline src/drivers/serial/raspi_uart.c]: Yes, you're right, I'll fix it in a moment

File: src/drivers/clock/raspi_systick.c (hunk 2 above)
Comment (anton-bondarev) [inline src/drivers/clock/raspi_systick.c]: Yes, it a chip constant. But not a board, another boards base on BCM2835 chip have the same base address. May be rename module as bcm2835_systick?

File: src/arch/arm/raspi/cpu_reset.S (hunk 3 above)
Comment (anton-bondarev) [inline src/arch/arm/raspi/cpu_reset.S]: Ok

File: src/drivers/clock/raspi_systick.c (hunk 2 above)
Comment (vloginova) [inline src/drivers/clock/raspi_systick.c]: Then I should rename all these modules a soon as the all part of RPi peripherals. Should I do it? It seems like there is no other board with bcm285 (at least popular, I couldn't find it).

Comment (anton-bondarev) [inline src/drivers/clock/raspi_systick.c]: Shouldn't. May be in future.
Ok.

**Pull request merged**